src/test/java/com/EighthLight/app/TestAi.java

```diff
@@ -70,21 +55,39 @@ void aiMakesTheRightMoveWithTwoSpotsLeft() {
         assertEquals(expectedBoardState, board.getSpaces());
     }
 
-//    @Test
-//    void makeMoveUpdatesTheBoardWithARandomMoveIfGameHasNotEnded() {
-//        String symbol = "X";
-//        String humanSymbol = "O";
-//        String difficulty = "Easy";
-//        ArrayList boardWithOpenSpace = new ArrayList(Arrays.asList("O", 1));
-//        MockBoard board = new MockBoard(boardWithOpenSpace, new ArrayList(), new ArrayList());
-//
-//        Ai computer = new Ai(symbol, humanSymbol, difficulty, board);
-//
-//        computer.makeMove(board);
-//
-//        assertEquals(1, board.getNumTimesGetSpacesCalled());
-//        assertEquals(1, board.getNumTimesUpdateSpaceIsCalled());
-//        assertEquals("1", board.getUpdateSpaceArgs().get(0));
-//        assertEquals(symbol, board.getUpdateSpaceArgs().get(1));
-//    }
+    @Test
+    void aiBlocksOpponentWhoIsAboutToWin() {
+        ArrayList boardState = new ArrayList(Arrays.asList("X", 1, 2,
+                                                           "X", "O", 5,
+                                                           6, 7, 8));
+
+        ArrayList expectedBoardState = new ArrayList(Arrays.asList("X", 1, 2,
+                                                                  "X", "O", 5,
+                                                                  "O", 7, 8));
+        Board board = new Board(3);
+
+        Ai ai = new Ai("O", "X", "Hard", board);
```

Comment:
When this class is fully tested, it may be confusing having all difficulty levels tested in one class/file